### PR TITLE
Yanntwd backend functions symmetricmatrices

### PR DESCRIPTION
**Pull request by ythanwerdas**

### CODE REVIEW

**Comment by ninamiolane** [comment]
I don't know why you have this travis test error while it's not your file, but the fix should be quick: let's not have atol=0.

File: geomstats/backend/numpy_linalg.py [hunk 1]
```diff
@@ -3,15 +3,86 @@
...
+            result = np.matmul(result, transp_eigvecs)
+        else:
+            result = np.vectorize(scipy.linalg.logm,
+                                  signature='(n,m)->(n,m)')(new_x)
+    else:
+        result = np.vectorize(scipy.linalg.logm,
+                              signature='(n,m)->(n,m)')(new_x)
+
+    if ndim == 2:
+        return result[0]
```
Comment (ninamiolane) [inline geomstats/backend/numpy_linalg.py]: Nice! This is linked to a discussion about shapes that we are having with Olivier, Niklas and Stefan

File: geomstats/backend/numpy_linalg.py [hunk 2]
```diff
@@ -3,15 +3,86 @@
...
+    if (new_x - np.transpose(new_x, axes=(0, 2, 1)) == 0).all():
+        eigvals, eigvecs = np.linalg.eigh(new_x)
+        eigvals = eigvals**power
+        if (eigvals > 0).all():
+            eigvals = np.vectorize(np.diag, signature='(n)->(n,n)')(eigvals)
+            transp_eigvecs = np.transpose(eigvecs, axes=(0, 2, 1))
+            result = np.matmul(eigvecs, eigvals)
+            result = np.matmul(result, transp_eigvecs)
+        else:
+            result = np.vectorize(scipy.linalg.logm,
```
Comment (ninamiolane) [inline geomstats/backend/numpy_linalg.py]: Change variable name to make it more explicit? there is `result` used 3 times

File: geomstats/backend/numpy_linalg.py [hunk 3]
```diff
@@ -3,15 +3,86 @@
...
+                                  signature='(n,m)->(n,m)')(new_x)
+            result = power * result
+            result = np.vectorize(scipy.linalg.expm,
+                                  signature='(n,m)->(n,m)')(result)
+    else:
+        result = np.vectorize(scipy.linalg.logm,
```
Comment (ninamiolane) [inline geomstats/backend/numpy_linalg.py]: same

File: tests/test_backend_numpy.py [hunk 4]
```diff
@@ -98,3 +98,40 @@ def test_expm_and_logm_vectorization(self):
...
+        power = .5
+        point = gs.array([[1., 0., 0.],
+                          [0., 4., 0.],
+                          [0., 0., 9.]])
+        result = gs.linalg.powerm(point, power)
+        expected = gs.array([[1., 0., 0.],
+                             [0., 2., 0.],
+                             [0., 0., 3.]])
+
+        self.assertTrue(gs.allclose(result, expected))
```
Comment (ninamiolane) [inline tests/test_backend_numpy.py]: Use assertAllClose (see geomstats/tests.py file)

File: tests/test_backend_numpy.py [hunk 5]
```diff
@@ -98,3 +98,40 @@ def test_expm_and_logm_vectorization(self):
...
+                            [0., 4., 0.],
+                            [0., 0., 9.]],
+                           [[1., 0., 0.],
+                            [0., 2.5, 1.5],
+                            [0., 1.5, 2.5]]])
+        result = gs.linalg.powerm(points, power)
+        result = gs.linalg.powerm(result, 1/power)
+        expected = points
+
+        self.assertTrue(gs.allclose(result, expected))
```
Comment (ninamiolane) [inline tests/test_backend_numpy.py]: same

File: tests/test_backend_numpy.py [hunk 6]
```diff
@@ -98,3 +98,40 @@ def test_expm_and_logm_vectorization(self):
...
+    def test_powerm(self):
+        power = 2.4
+        point = gs.array([[1., 0., 0.],
+                          [0., 2.5, 1.5],
+                          [0., 1.5, 2.5]])
+        result = gs.linalg.powerm(point, power)
+        result = gs.linalg.powerm(result, 1/power)
+        expected = point
+
+        self.assertTrue(gs.allclose(result, expected))
```
Comment (ninamiolane) [inline tests/test_backend_numpy.py]: same

File: geomstats/backend/numpy_linalg.py [hunk 7]
```diff
@@ -3,15 +3,86 @@
 import numpy as np
 import scipy.linalg
 
+from geomstats.backend.numpy import to_ndarray
+
+
+def exph(x):
```
[the remark this answers is not in the document]
Comment (ythanwerdas) [reply geomstats/backend/numpy_linalg.py]: h is for Hermitian, like in numpy.linalg.eigh
But I can name it expsym if you want

File: geomstats/backend/numpy_linalg.py (hunk 1 above)
Comment (ythanwerdas) [reply geomstats/backend/numpy_linalg.py]: Yes I discussed with them as well. I wonder if this trick works in any case. If yes, it should be applied in many cases according to them.

**Comment by ninamiolane** [comment]
I fix the atol problem in the parallel transport, just rebase on master ;)

**Pull request merged**